pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -482,52 +483,75 @@ func (r *MemberRollReconciler) reconcileNamespaces(ctx context.Context, namespac
 		return nil, nil, err, nil
 	}
 
-	for ns := range namespacesToRemove {
-		if ns == smmr.Namespace {
-			// we never operate on the control plane namespace
-			continue
-		}
-		err = reconciler.removeNamespaceFromMesh(ctx, ns)
-		if err != nil {
-			nsErrors = append(nsErrors, err)
-			pending.Insert(ns) // TODO: do we want to have a separate list for namespaces that are being removed, but haven't yet been?
-			r.EventRecorder.Event(
-				smmr,
-				corev1.EventTypeWarning,
-				string(eventReasonErrorRemovingNamespace),
-				fmt.Sprintf("Error removing namespace %q from mesh: %s", ns, err))
-		} else {
-			configured.Delete(ns)
-		}
+	reqLogger.Info("Reconciling member namespaces")
+	startTime := time.Now()
+
+	namespacesToProcess := namespacesToReconcile.Union(namespacesToRemove)
+	if namespacesToProcess.Has(smmr.Namespace) {
+		// we never operate on the control plane namespace
+		reqLogger.Info("ignoring control plane namespace in members list of ServiceMeshMemberRoll")
+		namespacesToProcess.Delete(smmr.Namespace)
+	}
+
+	// use scatter-gather pattern to process namespaces concurrently
+	type result struct {
+		ns string
+		err error
+	}
+	results := make(chan result, len(namespacesToProcess))
+
+	// scatter
+	for namespace := range namespacesToProcess {
+		go func(ns string) {
+			var err error
```

Comment:
I don't think it's necessary, as `WaitGroup` is nothing more than a counter, which we've already got (the messages in the channel). If the goroutine panics, the entire operator crashes, so it won't wait forever. The only scenario where it would is if we later add a `return` to the goroutine function without sending anything to the channel. Adding a `WaitGroup` would also not help in that case, as we'd then try to read the same number of results from the channel and end up waiting forever.

Reply:
oops, yeah that's right; panic() will stop the whole process. nevermind then